pjsip/src/pjsip/sip_auth_client.c

```diff
@@ -194,7 +194,11 @@ PJ_DEF(void) pjsip_auth_create_digest( pj_str_t *result,
 	digestNtoStr(digest, 16, ha1);
 
     } else if ((cred_info->data_type & PASSWD_MASK) == PJSIP_CRED_DATA_DIGEST) {
-	PJ_ASSERT_RETURN(cred_info->data.slen == 32, );
+	if (cred_info->data.slen != 32) {
+	    pj_bzero(result->ptr, result->slen);
+	    result->slen = 0;
+	}
+	PJ_ASSERT_RETURN(cred_info->data.slen == 32, PJ_EINVAL);
```

Comment:
minor, IMHO instead of using `PJ_ASSERT_RETURN()` outside the `if` block, perhaps it is easier to read to just use `pj_assert()` & `return EINVAL` inside the block? also a bit more efficient as the check is done once.